src/ui/public/accessibility/kbn_ui_ace_keyboard_mode.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * The `kbn-ui-ace-keyboard-mode` directive should be used on any element, that
+ * `ui-ace` is used on. It will prevent the keyboard trap, that ui-ace usually
+ * has, i.e. tabbing into the box won't give you any possibilities to leave
+ * it via keyboard again, since tab inside the textbox works like a tab character.
+ *
+ * This directive won't change anything, if the user uses the mouse. But if she
+ * tabs to the ace editor, an overlay will be shown, that you have to press Enter
+ * to enter editing mode, and that it can be left by pressing Escape again.
+ *
+ * That way the ui-ace editor won't trap keyboard focus, and won't cause that
+ * accessibility issue anymore.
+ */
+
+import angular from 'angular';
+import { uiModules } from 'ui/modules';
+import './kbn_ui_ace_keyboard_mode.less';
+import { ESC_KEY_CODE, ENTER_KEY } from 'ui_framework/services';
+
+uiModules.get('kibana').directive('kbnUiAceKeyboardMode', () => ({
+  restrict: 'A',
+  link(scope, element) {
+    const uniqueId = `uiAceKeyboardHint-${scope.$id}-${Math.floor(Math.random() * 10000)}`;
+
+    const hint = angular.element(`<div class="ui-ace-keyboard-hint" id="${uniqueId}" tabindex="0" role="application">
```

Comment:
`role=application` - I found it the best fitting role. This element kind of introduces it's own navigation schema (entering, exiting with keys) and also when in the editor, you won't be able to use tab anymore, and arrow keys will navigate all within that editor. But I also thought a lot what it should have, and figured out it might be the best (in contrast to textbox, which I mentioned I guess in the ticket somewhere, which would read out to a textbox, but apparently the user cannot do what she can do in any other textbox, since you require some additional enter press before editing starts).

Regarding the text: I can modify it like that. I actually find the text way more beautiful and had something similar in my first draft. The reason I decided for the very short sentence was: I fear people would press Enter directly when they hear "Press Enter to start editing. <short pause>" and miss the part, that they actually can exit by pressing Escape again, because the screen reader had a short pause between the sentences. That's why I tried to put it into one sentence. But I am glad to change it, if you feel like it's more reasonable. What do you think about people maybe missing the Escape mode?